### ISSUE TITLE
Kysely, better error messages for faulty migration?

### ISSUE THREAD

**Issue by lancejpollard** [koskimas/kysely#786]
I have a migration file with a few hundred lines in it, and I am getting this error:

```
$ pnpm tsx scripts/migrate.ts
failed to execute migration "0006"
failed to migrate
error: syntax error at or near ","
    at <anonymous> (./my_project/node_modules/.pnpm/pg@8.11.3/node_modules/pg/lib/client.js:526:17)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at PostgresConnection.executeQuery (./my_project/node_modules/.pnpm/kysely@0.26.3/node_modules/kysely/dist/cjs/dialect/postgres/postgres-driver.js:72:28)
    at <anonymous> (./my_project/node_modules/.pnpm/kysely@0.26.3/node_modules/kysely/dist/cjs/query-executor/query-executor-base.js:37:28)
    at SingleConnectionProvider.#run (./my_project/node_modules/.pnpm/kysely@0.26.3/node_modules/kysely/dist/cjs/driver/single-connection-provider.js:27:16)
    at PostgresConnection.executeQuery (./my_project/node_modules/.pnpm/kysely@0.26.3/node_modules/kysely/dist/cjs/dialect/postgres/postgres-driver.js:91:69)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at <anonymous> (./my_project/node_modules/.pnpm/kysely@0.26.3/node_modules/kysely/dist/cjs/query-executor/query-executor-base.js:37:28)
    at SingleConnectionProvider.#run (./my_project/node_modules/.pnpm/kysely@0.26.3/node_modules/kysely/dist/cjs/driver/single-connection-provider.js:27:16) {
  length: 90,
  severity: 'ERROR',
  code: '42601',
  detail: undefined,
  hint: undefined,
  position: '55',
  internalPosition: undefined,
  internalQuery: undefined,
  where: undefined,
  schema: undefined,
  table: undefined,
  column: undefined,
  dataType: undefined,
  constraint: undefined,
  file: 'scan.l',
  line: '1188',
  routine: 'scanner_yyerror'
}
```

How can I better debug what is going on?

`position: 55` doesn't seem to have anything to do with my migration file, nor does `line: 1188`, can I print something out which gives what postgres/kysely is reading internally perhaps somehow? Or how else can I figure out where the error is more easily? I've been resorting to commenting out lines of my migration to find the culprit, but then the partial migration succeeds, and I have to add _another_ migration file with the rest of the first file, and repeat. Would be great to avoid that somehow.

Thanks for the help!

I've boiled the error down to this migration:

```ts
import { Kysely } from 'kysely'

export async function down(db: Kysely<any>): Promise<void> {}

export async function up(db: Kysely<any>): Promise<void> {
  await db.schema
    .alterTable('files')
    .renameColumn('kind', 'sort')
    .addColumn('note', 'text')
    .addColumn('hold__code__base', 'uuid', c =>
      c.references('seed_file_hold.code__base').notNull(),
    )
    .addColumn('link__form__name', 'text', c => c.notNull())
    .addColumn('link__code__base', 'uuid', c => c.notNull())
    .execute()
}
```

But can't seem to idenitfy where that `,` comma is, but that is kind of beside the point, would be nice to know how to more effectively debug this.

After fixing that finally (I guess can't mix alterTable renames with adds), I get another similar error:

```
$ pnpm tsx scripts/migrate.ts
failed to execute migration "0006"
failed to migrate
error: column "reference" does not exist
    at <anonymous> (./my_project/node_modules/.pnpm/pg@8.11.3/node_modules/pg/lib/client.js:526:17)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at PostgresConnection.executeQuery (./my_project/node_modules/.pnpm/kysely@0.26.3/node_modules/kysely/dist/cjs/dialect/postgres/postgres-driver.js:72:28)
    at <anonymous> (./my_project/node_modules/.pnpm/kysely@0.26.3/node_modules/kysely/dist/cjs/query-executor/query-executor-base.js:37:28)
    at SingleConnectionProvider.#run (./my_project/node_modules/.pnpm/kysely@0.26.3/node_modules/kysely/dist/cjs/driver/single-connection-provider.js:27:16)
    at PostgresConnection.executeQuery (./my_project/node_modules/.pnpm/kysely@0.26.3/node_modules/kysely/dist/cjs/dialect/postgres/postgres-driver.js:91:69)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at <anonymous> (./my_project/node_modules/.pnpm/kysely@0.26.3/node_modules/kysely/dist/cjs/query-executor/query-executor-base.js:37:28)
    at SingleConnectionProvider.#run (./my_project/node_modules/.pnpm/kysely@0.26.3/node_modules/kysely/dist/cjs/driver/single-connection-provider.js:27:16) {
  length: 100,
  severity: 'ERROR',
  code: '42703',
  detail: undefined,
  hint: undefined,
  position: undefined,
  internalPosition: undefined,
  internalQuery: undefined,
  where: undefined,
  schema: undefined,
  table: undefined,
  column: undefined,
  dataType: undefined,
  constraint: undefined,
  file: 'tablecmds.c',
  line: '3527',
  routine: 'renameatt_internal'
}
```

So basically it would be nice to be able to better debug :)

**Comment by igalklebanov** [comment]
Hey 👋 

PostgreSQL's `alter table` allows either a single `rename column` action or multiple `add column` actions, not both. See https://www.postgresql.org/docs/current/sql-altertable.html.
